### PR TITLE
Extend OrderPromotion model and Promotions in Storefront API v2 Cart endpoints

### PR DESCRIPTION
**Pull request by damianlegawiec**

### CODE REVIEW

File: core/app/models/spree/order_promotion.rb [hunk 1]
```diff
@@ -2,5 +2,15 @@ module Spree
...
     belongs_to :promotion, class_name: 'Spree::Promotion'
+
+    delegate :name, :description, to: :promotion
+    delegate :currency, to: :order
+
+    extend Spree::DisplayMoney
+    money_methods :amount
+
+    def amount
+      order.all_adjustments.promotion.where(source: [promotion.actions]).sum(:amount)
```
Comment (piotrleniec-spark) [inline core/app/models/spree/order_promotion.rb]: Are these square brackets needed here? I think we can write `where(source: promotion.actions)`.

File: core/app/models/spree/order_promotion.rb [hunk 2]
```diff
@@ -2,5 +2,15 @@ module Spree
   class OrderPromotion < Spree::Base
     belongs_to :order, class_name: 'Spree::Order'
     belongs_to :promotion, class_name: 'Spree::Promotion'
+
+    delegate :name, :description, to: :promotion
+    delegate :currency, to: :order
+
+    extend Spree::DisplayMoney
```
Comment (piotrleniec-spark) [inline core/app/models/spree/order_promotion.rb]: Ruby style guide suggests to move all `include` and `extends` to the top of the class definitions (https://github.com/rubocop-hq/ruby-style-guide#consistent-classes).

File: core/spec/models/spree/order_promotion_spec.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+  let(:order) { create(:order_with_line_items) }
+  let(:promotion) { create(:promotion_with_item_adjustment, code: 'test') }
+
+  shared_context 'apply promo' do
+    before do
+      order.coupon_code = promotion.code
+      Spree::PromotionHandler::Coupon.new(order).apply
+      order.save!
+      order.all_adjustments.promotion.update_all(amount: -5.0)
```
Comment (piotrleniec-spark) [inline core/spec/models/spree/order_promotion_spec.rb]: Maybe we could use `adjustment_rate` transient field of `with_line_item_adjustment` factory to set this value (https://github.com/spree/spree/blob/master/core/lib/spree/testing_support/factories/promotion_factory.rb#L16)?

**Review by pl-spark — CHANGES_REQUESTED** [review]

File: core/app/models/spree/order_promotion.rb (hunk 2 above)
Comment (damianlegawiec) [reply core/app/models/spree/order_promotion.rb]: @piotrleniec-spark we're doing it differently for money - we like to keep it close-by (the extend and `money_methods` declaration)

File: core/spec/models/spree/order_promotion_spec.rb (hunk 3 above)
Comment (damianlegawiec) [reply core/spec/models/spree/order_promotion_spec.rb]: @piotrleniec-spark we want to test it explicitly here - this avoids any side-effect and false positives

**Review by pl-spark — APPROVED** [review]

**Pull request merged**